lib: fix strncpy usage

tomi-font opened the pull request:
Cherry picks from upstream PR https://github.com/OpenAMP/open-amp/pull/620.
This fixes `stringop-truncation` compilation errors.

bjarki-andreasen commented on open-amp/lib/remoteproc/remoteproc.c:
```diff
@@ -299,10 +299,12 @@ void remoteproc_init_mem(struct remoteproc_mem *mem, const char *name,
-	else
+	if (name) {
+		strncpy(mem->name, name, sizeof(mem->name) - 1);
+		mem->name[sizeof(mem->name) - 1] = 0;
```
minor nit, `0` or `'\0'` :)

tomi-font replied:
I personally prefer `'\0'`, but here there was already an instance of `= 0`, so that's why. :slightly_smiling_face:

arnopo committed:
lib: cherry pick strncpy() usage fix

Origin:
	https://github.com/OpenAMP/open-amp/pull/620

Commits:
	e233473d14654f08468595ad0dbe8f7e58acf267
	8591566382d055acd33f9d23e6826a8a4b0a1881
	1aecdc737d463b4ff1ece36847f4f2e68d4ffe4a

Status:
	Cherry pick PR that fixes the `stringop-truncation`
	compilation errors due to former usage of `strncpy()`.

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

tejlmand reviewed (changes requested):
Open-amp PRs cannot been directly on top of the Zephyr open-amp fork, because the open-amp project is being filtered (apps folder removed) and placed in a sub-folder named `open-amp`, thus any upstream open-amp PRs must be manually adjusted to work.
See also:
https://github.com/zephyrproject-rtos/open-amp/blob/master/README
https://github.com/zephyrproject-rtos/open-amp/tree/master

tomi-font commented:
Ah right, thanks for noticing. ~~Fixed that.~~ Closing the PR as you made another one that's already merged (would have been nice to also tell about this!).
